bpf/lib/conntrack.h

```diff
@@ -253,12 +261,20 @@ static __always_inline __u8 __ct_lookup(const void *map, struct __ctx_buff *ctx,
 				return CT_REOPENED;
 			}
 			break;
+
 		case ACTION_CLOSE:
-			/* RST or similar, immediately delete ct entry */
-			if (dir == CT_INGRESS)
+			/* If we got an RST and have not seen both SYNs,
+			 * terminate the connection
+			 */
+			if (!ct_entry_seen_both_syns(entry) &&
+			    (seen_flags.value & TCP_FLAG_RST)) {
 				entry->rx_closing = 1;
-			else
 				entry->tx_closing = 1;
+			} else if (dir == CT_INGRESS) {
+				entry->rx_closing = 1;
+			} else {
+				entry->tx_closing = 1;
+			}
```

Comment:
This should only be done for dir != CT_SERVICE since for the latter we may not see both directions.

Reply:
Do you mean that the last `else` should be `else if (dir == CT_EGRESS)` like so:

```
	/* If we got an RST and have not seen both SYNs,
	 * terminate the connection
	 */
	if (!ct_entry_seen_both_syns(entry) &&
	    (seen_flags.value & TCP_FLAG_RST)) {
		entry->rx_closing = 1;
		entry->tx_closing = 1;
	} else if (dir == CT_INGRESS) {
		entry->rx_closing = 1;
	} else if (dir == CT_EGRESS) {
		entry->tx_closing = 1;
	}
```

Reply:
```
if (!ct_entry_seen_both_syns(entry) && (seen_flags.value & TCP_FLAG_RST) && (dir != CT_SERVICE) {
    entry->rx_closing = 1;
    entry->tx_closing = 1;
} else if ...
```

Otherwise, we would prematurely mark the `CT_SERVICE` entry for removal. In the case of `CT_SERVICE`, we see packets only in one direction (to a service).

Reply:
Got it! Thanks. Should be fixed on the latest version.